src/optimizer/join_order/cardinality_estimator.cpp

```diff
@@ -160,10 +195,137 @@ double CardinalityEstimator::EstimateCrossProduct(const JoinNode &left, const Jo
+	}
+
+	auto table_indexes = op.GetTableIndex();
+	D_ASSERT(table_indexes.size() == 1);
+	idx_t table_index = table_indexes.at(0);
+
+	bool op_is_get = op.type == LogicalOperatorType::LOGICAL_GET;
+	bool op_is_proj = op.type == LogicalOperatorType::LOGICAL_PROJECTION;
+	LogicalProjection *proj;
+	LogicalGet *get;
+	if (op_is_get) {
+		get = &op.Cast<LogicalGet>();
+	}
+	if (op_is_proj) {
+		proj = &op.Cast<LogicalProjection>();
+	}
+
+	if (!(op_is_proj || op_is_get)) {
+		// Operation is probably a logical chunk get
+		if (op.type != LogicalOperatorType::LOGICAL_CHUNK_GET) {
+			// if operation is a logical chunk get, we can return. These are used to check
+			// if a column value is in a list of values. The list of values is the logical chunk
+			// get, and is usually not
+			throw InternalException("adding relation column mapping that is not a get or a projection");
+		}
+	}
+	for (idx_t it = 0; it < column_bindings.size(); it++) {
+		auto needed_binding_key = ColumnBinding(table_index, column_bindings.at(it).column_index);
+		if (needed_column_bindings.find(needed_binding_key) == needed_column_bindings.end()) {
+			// if the binding isn't used in any filters, no need to add it
+			continue;
+		}
+
+		auto key = ColumnBinding(relation_id, column_bindings.at(it).column_index);
+		ColumnBinding value;
+		string column_name = "";
+		if (op_is_get) {
+			value = ColumnBinding(table_index, get->column_ids.at(it));
+			if (get->column_ids.at(it) == DConstants::INVALID_INDEX) {
+				column_name = "rowid";
+			} else {
+				column_name = get->names.at(get->column_ids.at(it));
+			}
+		} else {
+			bool relation_is_constant = false;
+			if (op_is_proj) {
+				// need to be careful here. A projection can project an amount of columns not present in an original
+				// get. here we can get the name properly, but to get the column, you need to go and get the column
+				// index of the logical get from where the column is originating from. loop through each expression and
+				// find the bindings. probably needd to cast it here.
+				switch (proj->expressions[it]->type) {
+				case ExpressionType::BOUND_COLUMN_REF: {
+					auto &column_ref = proj->expressions[it]->Cast<BoundColumnRefExpression>();
+					value = column_ref.binding;
+					break;
+				}
+				case ExpressionType::BOUND_FUNCTION: {
+					// look for a bound column ref in the function children. If you find one
+					// add the binding for that column ref for statistics.
+					auto &tmp = proj->expressions[it]->Cast<BoundFunctionExpression>();
+					auto found_child_column_ref = false;
+					for (auto &child : tmp.children) {
+						if (child->type == ExpressionType::BOUND_COLUMN_REF) {
+							auto &column_ref = child->Cast<BoundColumnRefExpression>();
+							value = column_ref.binding;
+							found_child_column_ref = true;
+							break;
+						}
+					}
+					if (!found_child_column_ref) {
+						relation_is_constant = true;
+					}
+					break;
```

Comment:
add another case for operator cast. Again check the children for a bound column ref. 
add a test where you have something like 

select * from tbl1, tbl2 where CAST(tbl1.a as DOUBLE) = tbl2.b;